### PR TITLE
[Az.Tools.Installer]: Updates for a new minor/patch version

### LINKED ISSUE #19839: [Feature]: Improve error message for Az.Tools.Installer
### Description of the new feature

Error message with Az.Tools.Installer should be improved.
For example, when `-Repository` is missing: 

image

The error message is present at the beginning but then is buried with other error messages.
The modules should catch those errors and provide a simpler error message with a potential recommendation.

### Proposed implementation details (optional)

_No response_

### PR DESCRIPTION
**Pull request by matsest**
## Description

There hasn't been a release for Az.Tools.Installer for a long time (0.2.0) - even with changes being made to it several months ago -  so adding a few updates to hopefully make it ready for a new release.

- Set Repository parameter to Mandatory for `Install-AzModule` and `Update-AzModule` (fixes #19839)
- Update telemetry assembly to latest stable version
- Formatting of PS files
- Generate and update markdown/help files
- Add changelog file for this module

## Checklist

- [x] Check this box to confirm: **I have read the [_Submitting Changes_](../blob/main/CONTRIBUTING.md#submitting-changes) section of [`CONTRIBUTING.md`](../blob/main/CONTRIBUTING.md) and reviewed the following information:**

* **SHOULD** select appropriate branch. Cmdlets from Autorest.PowerShell should go to [`generation`](https://github.com/Azure/azure-powershell/tree/generation) branch. 
* **SHOULD** make the title of PR clear and informative, and in the present imperative tense. 
* **SHOULD** update `ChangeLog.md` file(s) appropriately
    * For any service, the `ChangeLog.md` file can be found at `src/{{SERVICE}}/{{SERVICE}}/ChangeLog.md`
    * A snippet outlining the change(s) made in the PR should be written under the `## Upcoming Release` header in the past tense. Add changelog in description section if PR goes into [`generation`](https://github.com/Azure/azure-powershell/tree/generation) branch.
    * Should **not** change `ChangeLog.md` if no new release is required, such as fixing test case only.
* **SHOULD** have approved design review for the changes in [this repository](https://github.com/Azure/azure-powershell-cmdlet-review-pr) ([_Microsoft internal only_](../blob/main/CONTRIBUTING.md#onboarding)) with following situations
    * Create new module from scratch
    * Create new resource types which are not easy to conform to [Azure PowerShell Design Guidelines](../blob/main/documentation/development-docs/design-guidelines)
    * Create new resource type which name doesn't use module name as prefix
    * Have design question before implementation
* **SHOULD** regenerate markdown help files if there is cmdlet API change. [Instruction](../blob/main/documentation/development-docs/help-generation.md#updating-all-markdown-files-in-a-module)
* **SHOULD** have proper test coverage for changes in pull request.
* **SHOULD NOT** introduce [breaking changes](../blob/main/documentation/breaking-changes/breaking-changes-definition.md) in Az minor release except preview version.
* **SHOULD NOT** adjust version of module manually in pull request

### CODE REVIEW

**Comment by Nickcandy** [comment]
Changing parameters to mandatory would cause a breaking change which is not allowed in regular releases. Is that right? @BethanyZhou

File: tools/Az.Tools.Installer/ChangeLog.md [hunk 1]
```diff
@@ -0,0 +1,10 @@
+## Upcoming Release
+* Make Repository parameter mandatory for `Install-AzModule` and `Update-AzModule`
```
Comment (BethanyZhou) [inline tools/Az.Tools.Installer/ChangeLog.md:2]: Please use paste tense for change log messages

**Comment by BethanyZhou** [comment]
> Changing parameters to mandatory would cause a breaking change which is not allowed in regular releases. Is that right? @BethanyZhou

That's true for module with stable version. Az.Tools.Installer is under preview, its version is 0.2.0, That's fine to include breaking changes.

File: tools/Az.Tools.Installer/ChangeLog.md [hunk 2]
```diff
@@ -0,0 +1,10 @@
...
+* Update telemetry assembly to 2.21.0
+* Fix typo in Az.Tools.Installer
```
Comment (matsest) [inline tools/Az.Tools.Installer/ChangeLog.md:4]:
```suggestion
* Made `Repository` parameter mandatory for `Install-AzModule` and `Update-AzModule`
* Updated telemetry assembly to 2.21.0
* Fixed typo in Az.Tools.Installer
```

Rewritten in past tense @BethandyZhou

File: tools/Az.Tools.Installer/Az.Tools.Installer.psd1 [hunk 3]
```diff
@@ -55,7 +55,7 @@
     
     # Assemblies that must be loaded prior to importing this module
     RequiredAssemblies = @(
-        '.\assembly\Microsoft.ApplicationInsights.2.12.0\Microsoft.ApplicationInsights.dll'
+        '.\assembly\Microsoft.ApplicationInsights.2.21.0\Microsoft.ApplicationInsights.dll'
```
Comment (matsest) [inline tools/Az.Tools.Installer/Az.Tools.Installer.psd1:58]: Let me know if this should **not** be updated (I see that it failed some tests in ADO, but I can't see where the relevant test is specified)

Comment (msJinLei) [reply tools/Az.Tools.Installer/Az.Tools.Installer.psd1]: > Let me know if this should **not** be updated (I see that it failed some tests in ADO, but I can't see where the relevant test is specified)
Hi @matsest, Welcome to contribute to Az.Tools.Installer！

The pipeline fails of other reasons. We are fixing it.
However, we still cannot upgrade ApplicationInsights to the latest version for PowerShell 7.0+ cannot support it. I have change the version to 2.13.1.

I also revert the change of `Repository` for we keep the parameter as optional to be consistent with PowerShellGet. To clarify the ambiguity, I add more explanation in the help message.

I remove the Az.Tools.Installer-help.xml for we will generate it every time we release a new version.

Please review the change I've made.

Comment (msJinLei) [reply tools/Az.Tools.Installer/Az.Tools.Installer.psd1]: @matsest Could you report the issue here https://github.com/Azure/azure-powershell/issues？ Thanks

Comment (matsest) [reply tools/Az.Tools.Installer/Az.Tools.Installer.psd1]: @msJinLei I would assume the issue already listed in the description of this PR is covering this sufficiently - see https://github.com/Azure/azure-powershell/issues/19839

Let me know if there should be created another one

Comment (msJinLei) [reply tools/Az.Tools.Installer/Az.Tools.Installer.psd1]: OK. I will check it later.

**Review by msJinLei — APPROVED** [review]

File: tools/Az.Tools.Installer/Az.Tools.Installer.psd1 (hunk 3 above)
Comment (matsest) [reply tools/Az.Tools.Installer/Az.Tools.Installer.psd1]: @msJinLei I see this was merged but the issue regarding the Repository parameter was not resolved as I suggested.  Https://github.com/Azure/azure-powershell/issues/19839 should be reopened..

Comment (msJinLei) [reply tools/Az.Tools.Installer/Az.Tools.Installer.psd1]: > @msJinLei I see this was merged but the issue regarding the Repository parameter was not resolved as I suggested. [Https://github.com/Azure/azure-powershell/issues/19839](Https://github.com/Azure/azure-powershell/issues/19839) should be reopened..

I created an issue for it https://github.com/Azure/azure-powershell/issues/20209 and will fix it soon. Thanks for reporting.

**Pull request merged**